crates/krator/src/manifest.rs

```diff
@@ -0,0 +1,32 @@
+use tokio::sync::watch::{channel, Receiver, Sender};
+
+#[derive(Clone)]
+/// Wrapper for `ObjectState::Manifest` type which reflects
+/// the latest version of the object's manifest.
+pub struct Manifest<T: Clone> {
+    rx: Receiver<T>,
+}
+
+impl<T: Clone> Manifest<T> {
+    /// Create a new Manifest wrapper from the initial object manifest.
+    pub fn new(inner: T) -> (Sender<T>, Self) {
+        let (tx, rx) = channel(inner);
+        (tx, Manifest { rx })
+    }
+
+    /// Obtain a clone of the latest object manifest.
+    pub fn latest(&self) -> T {
```

Comment:
Nit: Should this just return a borrow and allow the user to clone? I know it might not be worth it

Reply:
I think this probably returns a .clone() because borrows [hold a read lock](https://docs.rs/tokio/1.0.2/tokio/sync/watch/struct.Receiver.html#method.borrow) and could cause the sender to block, so returning a borrow here would force the user to know this and keep the borrow short lived.

Reply:
This is correct, it was one of the reasons for wrapping here to protect the user from this easy misuse of the reference.